chaco/scales/time_scale.py

```diff
@@ -285,48 +285,41 @@ def ticks(self, start, end, desired_ticks=None):
     def cal_ticks(self, start, end):
         """ ticks() method for calendar-based intervals """
 
+        # start and end are in seconds since Epoch, get naive datetimes
         try:
-            start = datetime.fromtimestamp(start)
+            start_dt = datetime.fromtimestamp(start)
         except ValueError:
-            start = datetime(MINYEAR, 1, 1, 0, 0, 0)
+            start_dt = datetime(MINYEAR, 1, 1, 0, 0, 0)
         try:
-            end = datetime.fromtimestamp(end)
+            end_dt = datetime.fromtimestamp(end)
         except ValueError:
-            end = datetime(MAXYEAR, 1, 1, 0, 0, 0)
+            end_dt = datetime(MAXYEAR, 1, 1, 0, 0, 0)
 
+        # get range of years of interest
+        # add 2 because of python ranges + guard against timezone shifts
+        # eg. 20000101 -> 19991231 because of local timezone, so end is 1999+2
+        years = range(start_dt.year, end_dt.year+2)
```

Comment:
Any `datetime` constructed with `MAXYEAR + 2` will raise a `ValueError`. We should guard against that here, `min(end_dt.year + 2, MAXYEAR)`, or on line 292.